torch_xla/csrc/ops/ops.cpp

```diff
@@ -863,6 +863,29 @@ NodePtr LogicalOr(const Value& input, const Value& other) {
       std::move(lower_fn));
 }
 
+NodePtr NanToNum(const Value& input, const Value& nan, const Value& posinf,
+                 const Value& neginf) {
+  auto lower_fn = [](const Node& node, LoweringContext* loctx) -> XlaOpVector {
+    xla::XlaOp xla_input = loctx->GetOutputOp(node.operand(0));
+    xla::XlaOp nan_replacement = loctx->GetOutputOp(node.operand(1));
+    xla::XlaOp posinf_replacement = loctx->GetOutputOp(node.operand(2));
+    xla::XlaOp neginf_replacement = loctx->GetOutputOp(node.operand(3));
+    const xla::Shape& input_shape = XlaHelpers::ShapeOfXlaOp(xla_input);
+    auto element_type = input_shape.element_type();
+    if (!xla::primitive_util::IsFloatingPointType(element_type)) {
```

Comment:
We probably should not leave this check so deep in the actual lowering code. We should return much earlier in `aten_xla_type.cpp`, I will comment there too.